### PR TITLE
Add a few recipes to the engineering techfab

### PR DESCRIPTION
**Pull request by arimah**
## About the PR
Adds a brand new lathe recipe, `SolarAssemblyFlatpack`, which can be combined with glass to make a solar panel or a solar tracker electronics board and glass to make a solar tracker. The engineering techfab can already print solar tracker boards and solar control panels, so why not enable it to also make solar panels? Note: this is not a "real" flatpack, as there is no solar panel electronics board, so we can't really use the Flatpacker 1001 for this.

Also adds a few dynamic recipes to the engifab:

- Experimental welding tool
- Jaws of life
- Shipyard RCD
- Shipyard RCD ammo

All of the above require research to unlock.

## Why / Balance
Make the engineering techfab more fit for purpose. :)

The material cost for the flatpack was balanced around a few things:

* You can currently buy a solar assembly crate for $500. That gets you six solar assembly flatpacks. (And that box sells for $795 at a depot – it should be increased to $1000 or something.)
* At 5 steel per assembly, a full stack of steel, plus some traces of silver and gold, gets you six assemblies. Then you need to add 2 glass per panel (hence why glass is not part of the lathe recipe). That's 4.5 kW of free-but-occasionally-unreliable power.
* I don't want these to be _too_ expensive in terms of materials, as then no one would ever bother making them. They're a lot of effort to set up, and a ridiculous material cost might discourage people from even experimenting, or lead them to buying the flatpacks instead. Hence no plasma, plasteel or other exotic materials.
* The 0.5 gold is inspired by how the Flatpacker 1001 requires that amount of gold for all of its flatpacking. It's clearly for maximum conductivity, right? :)
* I also don't want these to be too _cheap_ – going below 5 steel would make it way too easy to mass-produce infinite power.

## How to test
1. Spawn an engineering techfab
2. Insert materials
3. Print a solar assembly flatpack
4. Spawn an R&D server and an R&D computer
5. Insert a "research point disk [DEBUG, DO NOT MAP]" into the server
6. Unlock all of industrial
7. Print an experimental welding tool, jaws of life, shipyard RCD and shipyard RCD ammo

## Media
image

With all of industrial researched:
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
Negative.

**Changelog**
:cl:
- add: The engineering techfab can now print solar assembly flatpacks, industrial welders and advanced industrial welders by default, and, with research, advanced tools and RCDs.
- tweak: Solar assembly crates are more expensive. Maybe you can ask an engineer to print some for you instead!

### CODE REVIEW

File: Resources/Prototypes/Nyanotrasen/Entities/Structures/Machines/lathe.yml [hunk 1]
```diff
@@ -299,6 +300,10 @@
       - CableStack
       - CableMVStack
       - CableHVStack
+      - ShipyardRCD # Frontier
+      - ShipyardRCDAmmo # Frontier
+      - WelderExperimental # Frontier
```
Comment (whatston3) [inline Resources/Prototypes/Nyanotrasen/Entities/Structures/Machines/lathe.yml:305]: Something that might be a nice-to-have would be to put the industrial version of the welder in the static recipes.

**Comment by whatston3** [comment]
A small list of things that would be interesting to see here (curious of your thoughts):
* Advanced remote signaller (unless this is removed for a reason)
* Industrial welding tool and/or advanced industrial welding tool (would have to be static recipes, if supported - not currently researchable)

Any other power tools we're missing?  I can't think of any.

Apart from that, I think this is good - I'd also push for that solar crate value increase happening here, 1000 (even 1500?) seems reasonable.  Good feature.

**Comment by whatston3** [comment]
Also this feature should absolutely have a changelog.

**Comment by arimah** [comment]
> Advanced remote signaller (unless this is removed for a reason)

It's something you can research, right? I don't know how useful it is in practice. I could add it to the list of dynamic recipes easily enough, not sure if it should be static.

> Industrial welding tool and/or advanced industrial welding tool (would have to be static recipes, if supported - not currently researchable)

Ah, interesting thoughts! My personal take on that: both would be really nice to have, but maybe the advanced industrial welder should remain salvage loot (mostly – it's mapped on the Hammer because it's _so easy_ to find). Actually, maybe the fact it's _so easy_ to find makes it okay to add it... I don't know!

> I'd also push for that solar crate value increase happening here

Sure. It's easy enough to add.

**Comment by arimah** [comment]
I added two new recipes for the industrial welder and the advanced industrial welder. The material costs are a bit arbitrary, sort of a middle ground between the normal welder and the experimental one. I also put the advanced remote signaller in the dynamic recipes. Making it a static recipe might undermine research a bit. :)
Lastly, I increased the price of the solar assembly crate to $1500. No more selling at a profit.

**Comment by arimah** [comment]
I don't know how the fuck the Discovery shuttle ended up in here, but I'll unfuck that now.

**Comment by whatston3** [comment]
Changes seem reasonable to me, will double check briefly in-game.  Hope I'm not missing anything with the advanced signaller, and I like the idea of the standard industrial welder being a version being made under material constraints.

**Review by whatston3 — APPROVED** [review]
Pushed a comment for the price change to the solar assembly crate.

Looks good, leaving this open for any discussion on balance/gameplay implications I might be missing.

**Comment by Jalleo** [comment]
Putting this here but if you wish I can put this into the ideas on discord instead most of these are more stylistic roleplay cosmetic improvement suggestions that might assist on the engineering techfab getting used more.



Alongside that should they be able to build floodlights for when they are building larger projects?



The tile meister might be a good idea to and perhaps as a easier ability to get due to the fact that this would encourage engineering ships to serve for a more decorator role as well as providing equipment to ships.

Providing engi fabs with television boards might also help encourage people wanting to get televisions for their ships as well to assist reporters doing news broadcasts indirectly and/or service ships hosting boxing matches.

Fire extinguishers should probably be added too. 

The high cap and potentially the reactor cells might be a good addition too. I dont know if the Engineering techfabs can build better parts either via research I did a quick check by R&D and then filtered by parts on the techfab and it didn't show many options there. Might be they can but it doesn't show there.

**Comment by ThatOneGoblin25** [comment]
> * Industrial welding tool and/or advanced industrial welding tool (would have to be static recipes, if supported - not currently researchable)

Bear in mind when you print industrial or advanced welding tools they come already loaded with welder fuel.

**Comment by arimah** [comment]
> Bear in mind when you print industrial or advanced welding tools they come already loaded with welder fuel.

As does a regular welder, and that only costs 4 steel and takes 2 seconds to print. You can print that in all kinds of lathes!

As for @Jalleo's suggestions: interesting ideas, but for another PR. This PR was meant to add a few recipes that seemed like good, obvious fits - not to put all sorts of things into the engifab. Balancing is tricky.

**Comment by whatston3** [comment]
Reconfirmed as working.

**Pull request merged**